xarray/plot/plot.py

```diff
@@ -530,6 +530,10 @@ def newplotfunc(darray, x=None, y=None, figsize=None, size=None,
 
         _update_axes_limits(ax, xincrease, yincrease)
 
+        # Rotate dates on xlabels
+        if np.issubdtype(xval.dtype, np.datetime64):
+            plt.gcf().autofmt_xdate()
```

Comment:
Since the axis object is already known, better to use `ax.autofmt_xdate` than `plt.gcf()...`.

Reply:
`autofmt_xdate` is a `Figure` method so that won't work. I could set the rotation manually though...

Reply:
ah right, my mistake.

You should be able to use [`ax.get_figure()`](https://matplotlib.org/api/_as_gen/matplotlib.axes.Axes.get_figure.html#matplotlib.axes.Axes.get_figure) to get the figure. For some reason, `gcf()` seems error prone, but I don't really have any evidence for that.

Reply:
Sounds good to me. Changed the corresponding call in `plot.line()` too